### DISCUSSION TITLE
classes/class_collisionpolygon2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#152] [User-contributed notes]
# classes/class_collisionpolygon2d

Inherits: Node2D< CanvasItem< Node< Object A node that provides a polygon shape to a CollisionObject2D parent. Description: A node that provides a polygon shape to a CollisionObject2D parent and al...

https://docs.godotengine.org/en/stable/classes/class_collisionpolygon2d.html

**Comment by DylanLem** [comment]
If you want to instance these for actual collision detection (say, for a hitbox cone that increases/decreases in size), you have to update your ```polygon``` property within the ```_physics_process()``` method. this isn't stated ANYWHERE from what I can find

**Reply by Selus** [reply]
It shouldn't matter where you change the polygon, I think you had some different kind of problem.

**Reply by physcopanda** [reply]
I got the same problem, dynamically updating the polygon shape only works from physics_process

**Reply by viywolf** [reply]
Thank you so much this comment actually saved me, it really should be mentioned somewhere in the documentation

**Comment by inputrandomness** [comment]
Why does the CollisionPolygon2D class give a warning in its description about "CollisionShape2D" when the CollisionShape2D page does not give this same warning? Does the issue apply to both classes? And  should it not read "CollisionPolygon2D" on this page?

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/101107

**Comment by XMabbX** [comment]
Why does CollisionPolygon2D doesn't have debug color property like CollisionShape2D?

**Reply by Calinou** [reply]
In the editor, CollisionPolygon2D gives random colors to its polygons, so that they can be distinguished from each other (it decomposes the concave shape into several convex shapes). In the running project, only the outline is visible when **Visible Collision Shapes** is enabled. This would likely need to be refactored first before the color can be made configurable.

See https://github.com/godotengine/godot/issues/99935.

**Reply by XMabbX** [reply]
Thanks for the quick response!

**Comment by JimmyCushnie** [comment]
> Collision build mode. Use one of the BuildMode constants.

This doesn't explain at all what "build_mode" is. Clicking the link just takes you back to this page. Does anybody know what the heck Build Mode is?

**Reply by Calinou** [reply]
Check the updated documentation on the `latest` branch: https://docs.godotengine.org/en/latest/classes/class_collisionpolygon2d.html#enum-collisionpolygon2d-buildmode
